### ISSUE TITLE
Jenkins check failure repported as success

### ISSUE THREAD

**Issue by thomasleveil** [arachnys/cabot#505]
When setting up jenkins checks, I noticed cabot reports success for jenkins checks that failed due to technical issue (like wrong url for jenkins in the cabot conf).

![image](https://cloud.githubusercontent.com/assets/42300/25535683/44513d4e-2c38-11e7-8c17-04aa69e4e0eb.png)

**Comment by JeanFred** [comment]
Indeed, it’s a long standing bug − particularly dangerous when renaming jobs. I thought we had a ticket for it but it turns out not − thanks for filing it :)

**Comment by frankh** [comment]
```
            # If something else goes wrong, we will *not* fail - otherwise
            # a lot of services seem to fail all at once.
```

Hmm...looks like this was a feature?

**Comment by JeanFred** [comment]
I had a look to see if we could add more fine-grained messaging here, and possibly selectively fail on some clearly defined error (bad auth, bad address).

`python-jenkins` just throws `JenkinsException` in most situations so that’s  not really actionable.

We could add more steps before fetching info, eg [get_whoami](https://github.com/openstack/python-jenkins/blob/master/jenkins/__init__.py#L560) ; but that feels like additional hoops.

Thinking more, I think it would make more sense to have that when defining the JenkinsConfig − in the same way we do it for AlertPlugins
